charts/kube-prometheus-stack/values.yaml

```diff
@@ -2630,6 +2631,7 @@ prometheus:
   ##
   podDisruptionBudget:
     enabled: false
+    apiVersion: {}
```

Comment:
Should this not be a string? And now that I think about this, why do you need to be able to configure the PDB version?

Reply:
> And now that I think about this, why do you need to be able to configure the PDB version?

I was wondering that too, but we're already doing it on the chart:

- https://github.com/prometheus-community/helm-charts/blob/39299ac90a3c8bc9912ca7ca289788556b5aa007/charts/kube-prometheus-stack/templates/alertmanager/ingress.yaml#L10
- https://github.com/prometheus-community/helm-charts/blob/39299ac90a3c8bc9912ca7ca289788556b5aa007/charts/kube-prometheus-stack/templates/prometheus/ingress.yaml#L9

Reply:
Totally missed that! I see that this PR is trying to fix an issue when the chart isn't deployed using helm (https://github.com/prometheus-community/helm-charts/issues/3655). This looks like an edge case :/. The [_helpers.tpl](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/templates/_helpers.tpl) file uses the same logic to detect other api versions, so updating this logic means we would have to allow the same for all of the other apis.

This can be easily overcome using the parameter [`--kube-version`](https://helm.sh/docs/helm/helm_template/) on `helm template`. Leaving a comment with this info on the [issue](https://github.com/prometheus-community/helm-charts/issues/3655).

Reply:
Hi, thanks for the review.

Reply:
Hi, is there any chance to have this reviewed/approved? It will help us unblock some upgrades

Reply:
We might have something better for this. I see you have to duplicate the API version key on each of the workload PDBs, and you might end up with the same duplication if you need it for ingress or other type of resource. There is the `kubeTargetVersionOverride` root key that should address this.

What do you think of adding logic to respect this key? Then you only need to change a single key.

Reply:
Thanks for the input @GMartinez-Sisti. I think the actual key used is `kubeVersionOverride`, but for some reason it didn't work. I'll close the PR and issue, as we fixed it on the tooling side with a client version upgrade that covers the majority of usecases.

Reply:
Thank you for the feedback and understanding 👍